thefuck/rules/git_clone_missing.py

```diff
@@ -0,0 +1,33 @@
+'''
+Rule: git_clone_missing
+
+Correct missing `git clone` command when pasting a git URL
+
+```sh
+>>> https://github.com/nvbn/thefuck.git
+git clone https://github.com/nvbn/thefuck.git
+```
+
+Author: Miguel Guthridge
+'''
+from six.moves.urllib import parse
+
+
+def match(command):
+    script = command.script
+    parts = command.script_parts
```

Comment:
No need to care about Bash nor ZSH nor Fish, only `sh` and `cmd`/`pwsh`:
```suggestion
    # Check for a command not found error
    if which(command.script_parts[0]) or not (
        'not found' in command.output or 'is not recognized as' in command.output
    ):
        return False
    script = command.script
```
It's also marginally faster than using a generator with `any`.

Nit: declare variables only when necessary.